model/task_history.go

```diff
@@ -337,7 +334,7 @@ func (iter *taskHistoryIterator) GetChunk(v *Version, numBefore, numAfter int, i
 		{"$group": groupStage},
 		{"$sort": bson.M{task.RevisionOrderNumberKey: -1}},
 	}
-	agg := database.C(task.Collection).Pipe(pipeline)
+	agg := database.C(task.Collection).Pipe(pipeline).MaxTime(time.Minute)
```

Comment:
In the last PR we purposefully changed this to be larger, re some advice from John:
> I would set this slightly larger than the load balancer timeout (90 secs?) so that we don't accidentally kill something that might have finished

Reply:
done